### ISSUE TITLE
[python] Support Python 3.11 (requires numba 0.57.0rc1)

### ISSUE THREAD

**Issue by bkmartinjr** [single-cell-data/TileDB-SOMA#544]
Our setup.cfg says Python versions 3.7+ are supported.  The CI currently only tests 3.8 and 3.9.  We need to add to the matrix 3.7, 3.10 and 3.11

---

**Update as of 13-Mar-2023:** numba support (see numba/numba#8304) is still WIP.  Guesstimate: H123

**Comment by johnkerl** [comment]
Looks like we'll need to wait on Python 3.11. From #599:

At https://github.com/single-cell-data/TileDB-SOMA/actions/runs/3661926647/jobs/6190605155 we see `numba` is not yet supported in Python 3.11. cc @thetorpedodog @bkmartinjr @Shelnutt2 .

```
Collecting numba>=0.41.0
  Downloading numba-0.56.4.tar.gz (2.4 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.4/2.4 MB 146.6 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Running command python setup.py egg_info
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "/tmp/pip-install-0tpo_aju/numba_b1ec8a6b557b450ea19a316e9668617a/setup.py", line 51, in <module>
      _guard_py_ver()
    File "/tmp/pip-install-0tpo_aju/numba_b1ec8a6b557b450ea19a316e9668617a/setup.py", line 48, in _guard_py_ver
      raise RuntimeError(msg.format(cur_py, min_py, max_py))
  RuntimeError: Cannot install on Python version 3.11.0; only versions >=3.7,<3.11 are supported.
```

See also https://github.com/numba/numba/issues/8304

Looks like they're hoping for a release candidate by end of 2022: https://github.com/numba/numba/issues/8304#issuecomment-1336218610

**Comment by johnkerl** [comment]
#620 (merged) implements Python 3.7 support

Python 3.11 is still blocked on the aforementioned numba-dependency issue
